src/Service.php

```diff
@@ -136,9 +110,68 @@ public function getPermissions(Concrete $object, array $visitedIds = []): array
      * @param string $resource
      * @return bool
      */
-    public function isAllowed(Concrete $object, $resource): bool {
+    public function isAllowed(Concrete $object, $resource): bool
+    {
         $permissions = $this->getPermissions($object);
         return $permissions[$resource] == self::ALLOW;
     }
 
+    /**
+     * @param Concrete|Objectbrick\Data\AbstractData $object
+     * @param ClassDefinition\Data $fieldDefinition
+     * @param bool $ignoreObjectBrickFields
+     * @return array
+     */
+    protected function getPermissionsByFieldDefinition(
+        AbstractModel $object,
+        ClassDefinition\Data $fieldDefinition,
+        bool $ignoreObjectBrickFields = false
+    ): array {
+        $permissions = $permissionObjects = [];
+        switch (true) {
+            case ($fieldDefinition instanceof PermissionManyToManyRelation):
+                $permissionObjects = $object->get($fieldDefinition->getName());
+                break;
+
+            case ($fieldDefinition instanceof PermissionManyToOneRelation):
+                $manyToOneRelation = $object->get($fieldDefinition->getName());
+                if ($manyToOneRelation) {
+                    $permissionObjects = [$manyToOneRelation];
+                }
+                break;
+
+            case ($fieldDefinition instanceof PermissionResource):
+                $permissions = [$fieldDefinition->getName() => $object->get($fieldDefinition->getName())];
+                break;
+
+            case ($fieldDefinition instanceof DynamicPermissionResource):
+                $permissions = $object->get($fieldDefinition->getName());
+                break;
+
+            case (!$ignoreObjectBrickFields && $fieldDefinition instanceof ClassDefinition\Data\Objectbricks):
+                /* @var $objectBrick Objectbrick */
+                $objectBrick = $object->get($fieldDefinition->getName());
+                foreach ($objectBrick->getBrickGetters() as $brickGetter) {
+                    /* @var $brick Objectbrick\Data\AbstractData */
+                    $brick = $objectBrick->$brickGetter();
+                    if (!$brick) {
+                        continue;
+                    }
+
+                    $brickFieldDefinitions = $brick->getDefinition()->getFieldDefinitions();
+                    foreach ($brickFieldDefinitions as $brickFieldDefinition) {
+                        [$brickPermissions, $brickPermissionObjects] = $this->getPermissionsByFieldDefinition(
+                            $brick,
+                            $brickFieldDefinition,
+                            true
```

Comment:
This is where the parameter is used

Reply:
I saw that, was just wondering if it is really necessary... because bricks in bricks are not possible anyway, and so `$fieldDefinition instanceof ClassDefinition\Data\Objectbricks` never will be true in 'subcalls'. 
It would things make a bit easier to read, but I'm not sure if I miss something there

Reply:
As mentioned in my other comment, I think you're right. I'll remove it.